scipy/signal/_signaltools.py

```diff
@@ -492,10 +521,13 @@ def _freq_domain_conv(in1, in2, axes, shape, calc_fast_len=False):
         `in1` with `in2`.
 
     """
+    if xp is None:
+        xp = np
```

Comment:
if we use `_array_api.np_compat` instead of `np` here, does that remove the need for your changes to `is_complex`?